### PR TITLE
Feature learning flows

### PR DESCRIPTION
**Pull request by rakshasa**

### CODE REVIEW

File: integration_test/config/itest.yml [hunk 1]
```diff
@@ -30,7 +30,8 @@ dataset_options:
   public_router_ipv4_address: 172.16.90.4
   physical_network_simulated_ip: 10.210.0.2
   physical_network_vm1_ip: 10.210.0.104
-  physical_network_vm7_ip: 10.210.0.105
+  physical_network_vm5_ip: 10.210.0.105
```
Comment (Metallion) [inline integration_test/config/itest.yml]: What's this new vm5?

File: client/vnctl/lib/vnctl/cli/datapath.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ class Datapath < Base
       option :is_connected, :type => :boolean, :desc => "Flag that detemines if the datapath is connected or not."
       option :dpid, :type => :string, :desc => "Hexadecimal id for the datapath."
       option :node_id, :type => :string, :desc => "The node id for the datapath."
+      option :enable_ovs_learn_action, :type => :boolean, :desc => "Enables the datapath to use OVS learning flows."
```
Comment (Metallion) [inline client/vnctl/lib/vnctl/cli/datapath.rb]: How about adding `:default => true` here? I know it's already set as such in the database but I think it'll be more clear if set as such here too. Otherwise it gives me the feeling that I have to set this flag to enable it because the flag itself start with the word enable.

File: integration_test/dataset/wanedge.yml [hunk 3]
```diff
@@ -144,13 +160,18 @@ translations:
   - uuid: tr-saddr
     interface_uuid: if-glo2vnetgw
     mode: static_address
-    passthrough: true
+    passthrough: false
```
Comment (Metallion) [inline integration_test/dataset/wanedge.yml]: Why was this change needed?

File: vnet/lib/vnet/endpoints/1.0/vnet_api.rb [hunk 4]
```diff
@@ -115,6 +115,21 @@ def update_by_uuid(class_name, fill = {})
       respond_with(response.generate(updated_object))
     end
 
+    def update_by_uuid2(class_name, fill = {})
```
Comment (Metallion) [inline vnet/lib/vnet/endpoints/1.0/vnet_api.rb]: Maybe add a big ugly deprecation message in the log to the old `update_by_uuid`?

File: integration_test/config/itest.yml (hunk 1 above)
Comment (rakshasa) [inline integration_test/config/itest.yml]: It's for testing promiscuous and wanedge using GRE tunnels, while we used to only test with MAC2MAC.

Those tests involving vm5 are however marked as pending as there is an issue with GRE tunnels, and fixing that is outside of the scope of this pull request.

File: client/vnctl/lib/vnctl/cli/datapath.rb (hunk 2 above)
Comment (rakshasa) [inline client/vnctl/lib/vnctl/cli/datapath.rb]: The default value is in the database, and the webapi shouldn't imo pass down values to node_api that the user didn't explicitly supply. It means we'll have the default value defined in two different places.

File: integration_test/dataset/wanedge.yml (hunk 3 above)
Comment (rakshasa) [inline integration_test/dataset/wanedge.yml]: We're not testing passthrough and the correct setting for SNAT should be with no passthrough. Only addresses that have translations should pass.

File: vnet/lib/vnet/endpoints/1.0/vnet_api.rb (hunk 4 above)
Comment (rakshasa) [inline vnet/lib/vnet/endpoints/1.0/vnet_api.rb]: The old update/update_uuid refactoring in node_api isn't yet done.

When a class node_api is refactored then the above method needs to be used or the unit tests fail so logging doesn't really change anything.

**Comment by rakshasa** [comment]
https://github.com/axsh/wakame-ci-cluster/pull/9

Fixed diagram and replied to comments.

**Comment by Metallion** [comment]
:+1:
